eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/test/RecoverTestAccessioningConfiguration.java

```diff
@@ -0,0 +1,91 @@
+import uk.ac.ebi.ampt2d.commons.accession.hashing.SHA1HashingFunction;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.service.ContiguousIdBlockService;
+
+import uk.ac.ebi.eva.accession.core.model.ISubmittedVariant;
+import uk.ac.ebi.eva.accession.core.model.SubmittedVariant;
+import uk.ac.ebi.eva.accession.core.configuration.ApplicationProperties;
+import uk.ac.ebi.eva.accession.core.service.nonhuman.eva.SubmittedVariantAccessioningDatabaseService;
+import uk.ac.ebi.eva.accession.core.repository.nonhuman.eva.SubmittedVariantAccessioningRepository;
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantEntity;
+import uk.ac.ebi.eva.accession.core.service.nonhuman.eva.SubmittedVariantMonotonicAccessioningService;
+import uk.ac.ebi.eva.accession.core.summary.SubmittedVariantSummaryFunction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This configuration class has the single purpose of having loaded in MongoDB an object *before* the
+ * MonotonicAccessionGenerator is instantiated (and autowired in the accessioning service and pipeline jobs) so that
+ * the generator can recover from uncommitted accessions.
+ *
+ * An uncommitted accession is an accession that is present in MongoDB but wasn't committed in the block service (e.g.
+ * due to an unexpected crash of the application in previous executions). If the block service doesn't recover, this
+ * might lead to a single accession being assigned to several different objects in mongo.
+ */
+@Configuration
+public class RecoverTestAccessioningConfiguration {
+
+    private static final Logger logger = LoggerFactory.getLogger(RecoveringAccessioningConfiguration.class);
+
+    @Bean
+    public SubmittedVariantMonotonicAccessioningService submittedVariantMonotonicAccessioningService(
+            @Autowired @Qualifier("testSubmittedVariantAccessionGeneratorRecover")
+            MonotonicAccessionGenerator<ISubmittedVariant> accessionGenerator,
+            @Autowired SubmittedVariantAccessioningDatabaseService databaseService) {
+        return new SubmittedVariantMonotonicAccessioningService(accessionGenerator,
+                databaseService,
+                new SubmittedVariantSummaryFunction(),
+                new SHA1HashingFunction());
+    }
+
+    @Bean("testSubmittedVariantAccessionGeneratorRecover")
+    public MonotonicAccessionGenerator<ISubmittedVariant> testSubmittedVariantAccessionGeneratorRecover(
+            @Autowired SubmittedVariantAccessioningRepository repository,
+            @Autowired SubmittedVariantAccessioningDatabaseService databaseService,
+            @Autowired ApplicationProperties properties,
+            @Autowired ContiguousIdBlockService blockService) {
+
+        repository.deleteAll();
+
+        List<SubmittedVariantEntity> submittedVariantEntityList = new ArrayList<>();
+        for(long i=5000000000l;i<5000000030l;i++){
```

Comment:
Can you change the range slightly so that one block get committed partially ?
I would suggest adding a 3rd block 60->90 in the block allocation table and 1 or 2 submitted variant accession in this range. 
I'd be curious to see how the recover state deal with this situation.

Reply:
done